### ISSUE TITLE
Fix Combobox test errors with delite 0.7 beta

### ISSUE THREAD

**Issue by edchat** [ibm-js/deliteful#539]
Combobox is getting test errors after the updates to delite for the 0.7 beta.

These problems seem to be related to StoreMap, and the fact that this.attached is being set to true in CustomElement.attachedCallback.before, and this is happening before the call to StoreMap.queryStoreAndInitItems(), so it does not setup the _pendingQuery even though the StoreMap.attachedCallback() has not been called yet.

If the setting of this.attached is moved to the end of the before, or the beginning of the after in CustomElement.attachedCallback then the errors in the unit tests do not occur.

@wkeese What do you think?

**Comment by edchat** [comment]
Well, it looks like the change I mentioned above fixed the unit tests running locally, but I still see errors running on Saucelabs, so maybe there is also something else going on too.

**Comment by wkeese** [comment]
OK.  It seems a bit strange that `StoreMap#queryStoreAndInitItems()` is referencing `this.attached` but anyway as I wrote in #538, I agree that `attached` should be set after `attachedCallback()` completes since that's how it's documented.   I will check in that change.

**Comment by wkeese** [comment]
PS: Thanks for tracking this down.

**Comment by edchat** [comment]
@wkeese can you create another beta release with the latest delite updates?  I think that will help fix the deliteful errors in travis.  Thanks

**Comment by edchat** [comment]
I was able to see the Combobox unit test consistently fail locally with the iOS simlator, and if I add a call to:
'''combo.deliver()'''
here: https://github.com/ibm-js/deliteful/blob/master/tests/unit/Combobox.js#L136
Then all of the Combobox tests seem to pass, but I am not sure it makes sense that we would need to call deliver here.

**Comment by wkeese** [comment]
I removed a bunch of deliver() calls that were no longer needed (or at least, shouldn't be needed) in bfce99b92d83ad85912a50eb21c23c362d4c82d9.   Looking at the code more closely, I guess we need a deliver() call in initCombobox().   Or perhaps list.deliver()?

**Comment by wkeese** [comment]
PS: alternately,  create the Combobox in a more standard way using `new Combobox({...})` without setting `combo.list.store = ...` directly.

**Comment by edchat** [comment]
Some of the Combobox tests that are failing are not going through initCombobox(). that is only hit on the programmatic tests, so adding calls to deliver() there will not solve everything.

I am also seeing Combobox failures on IE10, and I am even having trouble with the Combobox sample on IE10, but it is failing with so many errors that I can not debug it in IE10 emulation mode.

**Comment by wkeese** [comment]
> Some of the Combobox tests that are failing are not going through initCombobox(). that is only hit on the programmatic tests, so adding calls to deliver() there will not solve everything.

You're saying that there's a problem on declarative widgets too?  Perhaps the Combobox code itself needs to be updated, and/or there's a bug in delite.

Actually, I did notice one thing: `CustomElement#attachedCallback()` calls deliver() in before advice rather than after advice, so `deliver()` executes before `Combobox#attachedCallback()`.  I will push a fix for that.

Also, probably Combobox#deliver() should call deliver() on the list.

If that doesn't fix it, maybe `Combobox#_initList()` should be calling `deliver()` on itself and on `this.list`.   Mainly because `Combobox#_initList()` could be called after `Combobox#attachedCallback()` has finished.  I'm not sure.

> I am also seeing Combobox failures on IE10, and I am even having trouble with the Combobox sample on IE10, but it is failing with so many errors that I can not debug it in IE10 emulation mode.

OK.  The sample is running fine for me on IE10/Win7.

**Comment by harbalk** [comment]
The last commit on delite 8734cd8a181d734fda051f411e24f8856fec7fa3 broke something when using PageableList.
I don't know really the details but nothing comes in the props list when computeproperties is executed
excepted the attached attribute. Also anything can be done with the list.

**Comment by edchat** [comment]
Yes, @wkeese it looks like we need to roll that change back, and after adding calls to deliver() in Combobox._initList(), I am still seeing some issues on IE10 and IE11, but only for programmatic and only for mouse navigation.  And a different issue on FF and Chrome which could be a testcase problem.

**Comment by wkeese** [comment]
> nothing comes in the props list when computeproperties is executed
excepted the attached attribute.

I'm not seeing that.   In pageable-prog-1.html, `computeProperties()` is called a number of times.   In the final call `props` just contains `attached`, but previous calls `props` contains lots of properties.

> Also anything can be done with the list.

Yes, I see that it's stuck with the loading icon.

> it looks like we need to roll that change [to delite] back

I don't see any reason to do that.  I think the change to delite is good.  Probably this is a bug in PageableList.   One suspicious piece of code I see in `PageableList#computeProperties()` is:

```js
if (this.pageLength > 0 && this.attached) {
```

That `if()` statement makes `computeProperties()` ignore important notifications.

**Comment by wkeese** [comment]
PS: Also, none of the code in PageableList #computeProperties() computes property values.   Rather, it adjusts the rendering.  Therefore it should be in refreshRendering() instead.

**Comment by edchat** [comment]
I have updated PageableList and Combobox with this change https://github.com/ibm-js/deliteful/commit/c0a0c1e7a5ed5cd035a8f4e1c97758ea8dc98e8b
@wkeese, I moved the code in PageableList.computeProperties to refreshRendering, but I ran into problems when trying to set this.list only after it was attached, so I did not make that change.  With these changes and my latest Combobox changes I think the Combobox tests should pass, but I will need an updated delite beta release before the travis tests will have the latest code needed.

**Comment by wkeese** [comment]
OK thanks.   I made an additional commit (0aad62e398996d4eb0dd8784e825d409b2e0c9e6) to do that last part.

I also tagged another delite beta, as you requested, although it's silly to put a new tag after every single change.
